text/353-cfn-registry-constructs.md

```diff
@@ -0,0 +1,467 @@
+as a separate module, with a version that corresponds to the extension version in the CloudFormation
+registry. This way, we will have 1:1 alignment and users will have the most freedom.
+
+#### Code Generator
+
+The code generator will be executed daily (e.g. through a GitHub workflow),
+query the CloudFormation Registry, and generate L1s for all the types based
+on their metadata and schema.
+
+We can use the `ListTypes` API to list all the types in the registry:
+
+```shell
+aws cloudformation list-types --visibility=PUBLIC
+```
+
+And then for each type, we can use `DescribeType` to retrieve the type information and its schema. 
+
+For example, this command will return the description of the `AWSQS::EKS::Cluster` resource:
+
+```shell
+aws cloudformation describe-type --arn arn:aws:cloudformation:us-east-1::type/resource/408988dff9e863704bcc72e7e13f8d645cee8311/AWSQS-EKS-Cluster
+```
+
+For reference: the output of this command can be found [here](https://gist.github.com/eladb/bf417c07444027d6954360295df4ee37#file-awsqs-vpc-vpcqs-module-json). 
+The parsed `Schema` field can be found [here](https://gist.github.com/eladb/bf417c07444027d6954360295df4ee37#file-awsqs-vpc-vpcqs-module-schema-json).
+
+Now, we need to filter all non-`AWS::` types (the `AWS::` types are actually L1s), and then generate 
+an L1 construct for each one. This includes:
+
+1. Generating the construct class that extends `CfnResource`. I think we can use submodules to 
+   represent the type namespace (`awsqs.eks` in this case), and the name of the resource as the construct name `Cluster`.
+2. Generate an `XxxProps` JSII struct for this construct based on the `Schema` field. 
+   This can be done using [`json2jsii`](https://github.com/cdklabs/json2jsii), which is the same 
+   tool we use to implemented the `import` command in CDK8s and CDKtf.
+
+
+### Is this a breaking change?
+
+Nope.
+
+### What are the drawbacks of this solution?
+
+> Describe any problems/risks that can be introduced if we implement this RFC.
+
+### What alternative solutions did you consider?
+
+#### `cdk import` versus library
+
+As mentioned above, an alternative approach would be to add support for `import` in the CDK CLI. The downsides of this approach (at this point) 
+are that (a) it is a bigger investment as it involves a change to the CDK CLI; and (b) the Construct Hub won't be able to surface these
+types automatically (we will need some custom support for "imported" types, which is something that we would like to add to the Construct Hub
+in the future for all CDK domains).
+
+#### Monolithic versus module per resource
+
+As mentioned above, due to the fact that resource schemas are versioned and we want to allow users to select which
+version of the resource they use, we determined that the best approach is to publish a module for each resource.
+This has logicical/operational implications that we believe can mostly be mitigated using projen's capabilities to
```

Comment:
Can you be a bit more specific about the implications you're thinking about, because I don't really see them.

It seems to me we need a cron job either way, and publishing is mostly the same to what we're already doing in CDK; the only difference would really be whether we hand-write a single `package.json`, or generate many of them automatically... correct?

The projen path would expect us to commit the `package.json`s, which I'm not even sure we necessarily need to do... but I guess we can.

Reply:
Removed the sentence about projen and added "generated" to indicate that the modules will be generated. That's the important point. We will determine how to generate them later. That's not that important.